app/gui2/src/stores/graph.ts

```diff
@@ -354,9 +356,49 @@ export const useGraphStore = defineStore('graph', () => {
     stopCapturingUndo,
     updateNodeRect,
     updateExprRect,
+    executionMode,
   }
 })
 
+type ExecutionMode = 'live' | 'design'
+type Settings = { executionMode: WritableComputedRef<ExecutionMode> }
+function setupSettings(proj: ReturnType<typeof useProjectStore>): Settings {
+  const settings = computed(() => proj.module?.doc.settings)
+  // Value synchronized with a key of the `settings` map, used to enforce reactive dependencies.
+  const executionMode_ = ref<ExecutionMode>()
+  const executionMode = computed<ExecutionMode>({
+    get() {
+      return executionMode_.value ?? 'design'
+    },
+    set(value) {
+      // Update the synchronized map; the change observer will set `executionMode_`.
+      if (settings.value != null) settings.value.set('executionMode', value)
+    },
+  })
+  useObserveYjs(settings, (event) => {
```

Comment:
`useObserveYjs` is not enough to properly synchronize the value, as it only notifies about Yjs deltas, ignoring the observed object itself being changed. You also have to reassign it in case the whole settings object is swapped (e.g. by switching a project/module). You can add a `watch` over the `settings` value, and read its value (possibly using Yjs `toJSON` method) at the time it changes. The same thing can also be used for value initialization by declaring the watcher as `immediate`.

I'm also debating whether or not we should use some abstraction to not do that synchronization manually in each case. So far it wasn't really necessary, but it's worth to consider using https://syncedstore.org/docs/ if we end up adding more flags/values like that. We would have to check the bundle size cost though.

Reply:
> You also have to reassign it in case the whole settings object is swapped (e.g. by switching a project/module).

I moved it to the project model, so switching modules won't make a difference. I don't see how we would need to handle switching projects here--when a new project is opened we start a whole new GraphEditor with its own project store, right?

Reply:
Right now yes, but that assumption may not be correct in the future once we integrate the GUI with dashboard more closely.